<?xml version="1.0"?>
<launch>
  <arg name="device_ip" default="192.168.1.200" />
  <arg name="msop_port" default="2368" />
  <arg name="difop_port" default="2369" />
  <arg name="return_mode" default="1" />
  <arg name="time_synchronization" default="false" />


  <node pkg="lslidar_c16_driver" type="lslidar_c16_driver_node" name="lslidar_c16_driver_node" output="screen">
    <!--param name="pcap" value="$(find lslidar_c16_driver)/pcap/xxx.pcap" /-->
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>

    <!-- <param name="frame_id" value="laser_link"/> -->

     <param name="frame_id" value="velodyne"/>

    <param name="add_multicast" value="false"/>
    <param name="group_ip" value="224.1.1.2"/>
    <param name="rpm" value="600"/>
    <param name="return_mode" value="$(arg return_mode)"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
  </node>

  <node pkg="lslidar_c16_decoder" type="lslidar_c16_decoder_node" name="lslidar_c16_decoder_node" output="screen">
    <!-- <param name="scan_frame_id" value="laser_link"/> -->


    <param name="min_range" value="0.15"/>
    <param name="max_range" value="150.0"/>
    <param name="frequency" value="10"/>
    <param name="config_vert" value="true"/>
    <param name="print_vert" value="false"/>
    <param name="return_mode" value="$(arg return_mode)"/>
    <param name="degree_mode" value="2"/>
    <param name="config_vert_file" value="false"/>
    <param name="distance_unit" value="0.25"/>
    <param name="time_synchronization" value="$(arg time_synchronization)"/>
    <param name="scan_start_angle" value="0.0"/>
    <param name="scan_end_angle" value="36000.0"/>
    <param name="scan_num" value="8"/>
    <param name="publish_scan" value="true"/>
    <remap from="/scan_topic" to="scan"/>
    <param name="echo_second" value="false"/>
    
    <param name="pointcloud_topic" value="velodyne_points"/>
    <param name="coordinate_opt" value="true"/>
  </node>
 <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lslidar_c16_decoder)/rviz_cfg/c16.rviz" />  -->
 
  <!-- <node pkg="tf" type="static_transform_publisher" name="laser_link_to_world" args="0 0 0 0 0 0 world laser_link 100" /> -->
  
</launch>

